### ISSUE TITLE
PHP-FPM7.2:  ERROR: Unable to create the PID file

### ISSUE THREAD

**Issue by pc-magas** [php-earth/alpine#2]
I have installed the `php7-fpm` package from the repositories you provide in an apline 3.7 GNU/Linux virtual machine. When I try to launch `php-fpm` for some reason I get the following error:

```
[05-May-2018 13:14:14] ERROR: Unable to create the PID file (/run/php/php-fpm7.2.pid).: No such file or directory (2)
[05-May-2018 13:14:14] ERROR: FPM initialization failed
```
I receive the response if I run the following command:

```
sudo php-fpm
```
Furthermore I launched it as a service via:

```
sudo /etc/init.d/php-fpm7.2 start
```

But when I try to look for the runninc process it does not find it as the following commands show:

```
webserver:~$ ps -A | grep php-fpm
 2120 user       0:00 grep php-fpm
webserver:~$ ps -A | grep php
webserver:~$ ps -A | grep php*
 2128 user       0:00 grep php*
webserver:~$ ps -A | grep -e "/php.*/"
 2132 user       0:00 grep -e /php.*/
webserver:~$ ps -A php*
PID   USER     TIME   COMMAND
 2022 user       0:00 sshd: user@pts/0
 2023 user       0:00 -bash
 2134 user       0:00 ps -A php*
```

Also if I look over the log file `/var/log/php/7.2/php-fpm.log` I get the following piece of log:

```
[05-May-2018 13:21:18] NOTICE: configuration file /etc/php/7.2/php-fpm.conf test is successful

[05-May-2018 13:21:18] ERROR: Unable to create the PID file (/run/php/php-fpm7.2.pid).: No such file or directory (2)
[05-May-2018 13:21:18] ERROR: FPM initialization failed
```

Do you know why `php-fpm` fails to launch?

**Comment by pc-magas** [comment]
Further investigation shown that running it just after installation via apk works fine but on reboot does not work.

**Comment by petk** [comment]
Thanks for reporting. Cause is the wrong location of the pid file instead of
`/run/php/php-fpm7.2.pid`
it is set to be
`pidfile="/run/$RC_SVCNAME/php-fpm7.2.pid"` in the initd file... Patch coming up. :+1:

**Comment by petk** [comment]
Fixed via 5900e86f15b85de50a034904c4ac678c0aa878ba In case there are still some of the issues with initd scripts, let me know please. This is mainly used in a container environment so far :)
